iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/async/IoTDBDataRegionAsyncConnector.java

```diff
@@ -539,9 +539,20 @@ public synchronized void close() {
       tabletBatchBuilder.close();
     }
 
+    decreaseOrRemoveAsyncClient();
+
     super.close();
   }
 
+  private void decreaseOrRemoveAsyncClient() {
+    clientManager.decreaseAttributesCount();
+
+    if (clientManager.getAttributesCount() == 0) {
+      clientManager.removeAttributesCount();
+      clientManager.closeAndRemoveClientManager();
+    }
```

Comment:
This function is best placed in `IoTDBDataNodeAsyncClientManager` because the `put` code is in `IoTDBDataNodeAsyncClientManager`, and the `remove` code is also best placed in `IoTDBDataNodeAsyncClientManager`, which can maintain a better code structure. And the code in this function needs to be synchronized using the `IoTDBDataRegionAsyncConnector` object lock